src/main/java/core/AbstractPlayer.java

```diff
@@ -10,7 +10,7 @@
 public abstract class AbstractPlayer {
 
     // ID of this player, assigned by the game
-    int playerID;
+    protected int playerID;
```

Comment:
This change keeps creeping back :D but it should be package-private and not protected so that the game assigns it and players can only access it and not change it.

Reply:
Ooops. I was being sloppy to simplify my unit tests. I will go and complexify them.